scripts/des_y3_sne_ia/des_y3_sne_ia.py

```diff
@@ -0,0 +1,190 @@
+                'time': Value('float32'),
+            }),
+        }
+
+
+        # Adding all values from the catalog
+        for f in _FLOAT_FEATURES:
+            features[f] = Value("float32")
+        for f in _STR_FEATURES:
+            features[f] = Value("string")
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=Features(features),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """We handle string, list and dicts in datafiles"""
+        if not self.config.data_files:
+            raise ValueError(
+                f"At least one data file must be specified, but got data_files={self.config.data_files}"
+            )
+        data_files = dl_manager.download_and_extract(self.config.data_files)
+        if isinstance(data_files, (str, list, tuple)):
+            files = data_files
+            if isinstance(files, str):
+                files = [files]
+            # Use `dl_manager.iter_files` to skip hidden files in an extracted archive
+            files = [dl_manager.iter_files(file) for file in files]
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN, gen_kwargs={"files": files}
+                )
+            ]
+        splits = []
+        for split_name, files in data_files.items():
+            if isinstance(files, str):
+                files = [files]
+            # Use `dl_manager.iter_files` to skip hidden files in an extracted archive
+            files = [dl_manager.iter_files(file) for file in files]
+            splits.append(
+                datasets.SplitGenerator(name=split_name, gen_kwargs={"files": files})
+            )
+        return splits
+
+    def _generate_examples(self, files):
+        """Yields examples as (key, example) tuples."""
+        for file_number, file in enumerate(itertools.chain.from_iterable(files)):
+            with h5py.File(file, "r") as data:
+
+                keys = [data["object_id"][()]]
+
+                # Preparing an index for fast searching through the catalog
+                sort_index = np.argsort(data["object_id"][()])  # Accessing the scalar index
+                sorted_ids = [data["object_id"][()]]  # Ensure this is a list of one element
+
+                for k in keys:
+                    # Extract the indices of requested ids in the catalog
+                    i = sort_index[np.searchsorted(sorted_ids, k)]
+                    # Parse data
+                    idxs = np.arange(0, data["flux"].shape[0])
```

Comment:
i don't see `i` used anywhere in the `for k in keys` loop so i'm kind of confused since `i` is the object id we're trying to load in to `example`?

Reply:
I think this is due to there being only one example per hdf5 file, whereas this code has been inherited from other datasets where there are multiple examples per hdf5 file (in which case the index is used to select the particular examples required from the file). It should therefore be safe to remove `sort_index`, `sorted_ids`, and `i`. In future, we may want to switch to storing multiple examples per file and restore these variables.

Reply:
Actually we need to do a bit more than this, otherwise object_ids, k, and keys become unused variables, and we aren't able to selectively return data based on the provided object_ids. A quick fix would be to just iterate over those files that are requested as object_ids (or all files if no object_ids provided).

Reply:
This has now been fixed for all five surveys matching @tom-hehir solution in https://github.com/AstroPile/AstroPile_prototype/blob/3d6da624b945b1c993d8447042a23e4453ccaac0/scripts/yse/yse.py